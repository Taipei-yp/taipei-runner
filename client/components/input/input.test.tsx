import React from "react";
import TestRenderer, { act } from "react-test-renderer";
import { Input } from ".";

describe("Component Input", () => {
  const onChange = jest.fn();
  const input = TestRenderer.create(
    <Input
      value="testvalue"
      name="testname"
      className="testclass"
      fullWidth
      onChange={onChange}
    />,
  );

  it("correct render", () => {
    expect(input.toJSON()).toMatchSnapshot();
  });

  it("correct props", () => {
    expect(input.root.props.className).toBe("testclass");
  });

  it("onchange work", () => {
    act(input.root.props.onChange);
    expect(onChange).toBeCalledTimes(1);
  });
});
