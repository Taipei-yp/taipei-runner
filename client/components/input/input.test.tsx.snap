// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Input correct render 1`] = `
<input
  className={
    Object {
      "has": [Function],
      "is": [Function],
      "mix": [Function],
      "split": [Function],
      "state": [Function],
      "toString": [Function],
    }
  }
  name="testname"
  onChange={[MockFunction]}
  type="text"
  value="testvalue"
/>
`;
